lib/play.js

```diff
@@ -1,74 +1,175 @@
   return new Promise((resolve) => {
-    store.ws.on("message", (message) => {
+    player.ws.on("message", (message) => {
       let msg = JSON.parse(message);
-      if (msg.type === "prompt") {
-        return resolve([msg.players, msg.team]);
+      if (msg.type === "mode") {
+        return resolve(msg.mode);
       }
     });
   });
 };
 
-const gameSetup = async (store) => {
-  if (store.ws) {
-    const choices = await recieveGameSetupChoices(store);
-    store.setPlayers(choices[0], choices[1]);
-  } else {
-    const singlePlayer = await getNumPlayersChoice(store);
-    const teamChoice = await getTeamChoice(store);
-    store.setPlayers(singlePlayer, teamChoice);
-  }
+// const setGameMode = async (game, player) => {
+//   let mode;
+//   if (player.ws) {
+//     mode = await receiveGameModeChoice(player);
+//   } else {
+//     mode = await getGameModeChoice();
+//   }
+//   game.setMode(mode);
+// };
+
+const getGameMode = async (player) => {
+  return player.ws
+    ? await receiveGameModeChoice(player)
+    : await getGameModeChoice();
+};
+
+// const createPlayerArray = async (lobby, mode, player) => {
+//   if (mode === "cpu") {
+//     player.setName("You");
+//     return [player, new CpuPlayer()];
+//   } else if (mode === "local") {
+//     player.setName("Player 1");
+//     return [player, new HumanPlayer("Player 2")];
+//   } else if (mode === "online") {
+//     return await lobby.matchWithWaitingPlayer(player);
+//   }
+// };
+
+const assignPlayers = async (game, players) => {
+  const teams = randomTeams();
+  game.setPlayer(players[0], teams[0]);
+  game.setPlayer(players[1], teams[1]);
 };
 
-const play = async (store) => {
-  await gameSetup(store);
-  setPlayOrder(store, pickOrder());
-  while (!store.winner) {
-    const check = await checkForGameOver(store);
-    if (!check) {
-      await makeTurn(store);
-      store.setNextPlayer();
+const play = async (lobby, game, players) => {
+  if (players) {
+    await assignPlayers(game, players);
+    game.setPlayOrder();
+    while (!game.winner) {
+      const check = await checkForGameOver(game);
+      if (!check) {
+        await makeTurn(game);
+      }
     }
+  } else {
+    await waitForGameOver(game);
   }
 };
 
-const runPlayLoop = async (ws) => {
+const runPlayLoop = async (lobby, player) => {
+  console.log("Starting runPlayLoop");
   let playAgainResult = true;
   while (playAgainResult) {
-    const store = new Store(ws);
-    await play(store);
-
-    playAgainResult = await playAgain(store);
+    const mode = await getGameMode(player);
+    let game;
+    if (mode === "online") {
+      lobby.addAsWaitingPlayer(player);
+      let players = await lobby.waitForOpponent(player);
+      if (players) {
+        const game = new Game();
+        lobby.addGame(game);
+        await play(lobby, game, players);
+      } else {
+        const game = lobby.getGameFromPlayer(player);
+        await play(lobby, game);
```

Comment:
just `await waitForGameOver(game)` here, and remove the if/else from play() because it will always have players?